player/lua/ytdl_hook.lua

```diff
@@ -815,7 +815,7 @@ function run_ytdl_hook(url)
     local json, err = utils.parse_json(json)
 
     if (json == nil) then
-        msg.error("failed to parse JSON data: " .. err)
+        msg.error("failed to parse JSON data: " .. (err or "null json"))
```

Comment:
I'd prefer a more correct message, for instance, something like this maybe?
```lua
    if (json == nil) then
        if err then
            msg.error("failed to parse JSON data: " .. err)
        else
            msg.error("null JSON data")
        end
```

Reply:
On second thought, this separate block for json parsing errors shouldn't even exist. I'll move the parsing up so its errors can be handled with the others.

Reply:
Not sure what you mean, show me the code...